app/services/application/FinalOutcomeService.scala

```diff
@@ -78,19 +79,19 @@ trait FinalOutcomeService {
           for {
             (candidate, contactDetails) <- retrieveCandidateDetails(app.applicationId)
             _ <- emailClient.notifyCandidateOnFinalFailure(contactDetails.email, candidate.name)
-            progressStatuses <- applicationRepo.getLatestProgressStatuses
+            progressStatuses <- applicationRepo.getProgressStatusTimestamps(app.applicationId)
```

Comment:
Let's call this progressStatusTimestamps, fetch the last here, extra the status and pass it to progressToNotified.

Reply:
I changed the impl. to sort the progress statuses first in date time order and then get the last one.